src/common/dto/utils/DoesNotMatch.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from 'class-validator';
+
+/**
+ *  Compare the current field against another in the same DTO to ensure that the
+ * two do not match.
+ *
+ * @param property The field to compare against
+ * @param validationOptions ValidationOptions
+ */
+export default function DoesNotMatch<DTO>(
+  property: string & keyof DTO,
+  validationOptions?: ValidationOptions
+): PropertyDecorator {
+  return function DoesNotMatchValidator(
+    object: DTO,
+    propertyName: string & keyof DTO
+  ): void {
+    registerDecorator({
+      name: 'doesNotMatch',
+      target: object.constructor,
+      propertyName,
+      constraints: [property],
+      options: validationOptions,
+      validator: {
+        validate(value: string, args: ValidationArguments): boolean {
+          const target = args.object as DTO;
+          const [otherProp] = args.constraints as (string & keyof DTO)[];
+          const otherProperty = target[otherProp];
+          if (typeof otherProperty === 'string') {
```

Comment:
This makes sense for the particular `sameAs` use case, but for a generic validator we probably want to be able to check for types (e.g. numbers, booleans) right?

If not, we probably want to make it clear that it will only match strings.

Reply:
Yes, I did wonder about this at the time, but somehow only comparing strings made sense. I think you're right though. Removed that in 1679e590